File: Week2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("ERROR\n");
        return 1;
    }
    
    int key = atoi(argv[1])%26;

    printf("plaintext: ");
    string plaintext = get_string();
    
    printf("ciphertext: ");
    
    for (int i = 0, n = strlen(plaintext); i < n; i++)
    {
        if (!isalpha(plaintext[i]))
        {
            printf("%c", plaintext[i]);
        }
        else if (plaintext[i] <= 'z') 
        {
            if (plaintext[i] + key > 'z') 
            {
                printf("%c", plaintext[i] + key - 26);
            }
            else 
            {
                printf("%c", plaintext[i] + key);
            }
        }
        else if (plaintext[i] <= 'Z') 
        {
            if (plaintext[i] + key > 'Z') 
            {
                printf("%c", plaintext[i] + key - 26);
            }
            else
            {
                printf("%c", plaintext[i] + key);
            }
        }
    }
    printf("\n");
}

File: pset6/caesar.py
# branko stankovic
# cs 50 edX course
# https://branko-stankovic.github.io/

# cs50 library
import cs50
# system library for using arguments via sys.argv[]
import sys

# ensue proper user input
if len(sys.argv) != 2:
    print("Usage: ./caesar k")
    exit(1)

# remember the second argument as a key for ciphering
key = int(sys.argv[1])

# get plaintext from the user
print("plaintext: ", end="")
plaintext = cs50.get_string()

# set up for the ciphertext output
print("ciphertext: ", end="")

# start crypting the message!
for i in range(len(plaintext)):
    # if it is a letter
    if plaintext[i].isalpha():
        # and if it is uppercase
        if plaintext[i] <= "Z":
            # check does it go over the limit when ciphered
            if ord(plaintext[i]) + key > 90:
                print(chr(ord(plaintext[i])+key-26), end="")
            # if not just cipher it    
            else:
                print(chr(ord(plaintext[i])+key), end="")
        # and if it is lowercase  
        elif plaintext[i] <= "z":
            # check does it go over limit when ciphered
            if ord(plaintext[i]) + key > 122:
                print(chr(ord(plaintext[i])+key-26), end="")
                #print("#", end="")
            # if not, cipher it    
            else:
                print(chr(ord(plaintext[i])+key), end="")
                #print("*", end="")
    else:
        print(plaintext[i], end="")
print("")        